src/lua/oltp_common.lua

```diff
@@ -270,6 +292,60 @@ CREATE TABLE sbtest%d(
    end
 end
 
+function bulk_inserts(con, table_num)
+
+   if (sysbench.opt.table_size > 0) then
+      time = os.date("*t")
+   end
+   iquery = ""
+
+   if sysbench.opt.auto_inc then
+      iquery = "INSERT INTO sbtest" .. table_num .. "(k, c, pad) VALUES"
+   else
+      iquery = "INSERT INTO sbtest" .. table_num .. "(id, k, c, pad) VALUES"
+   end
+
+   con:bulk_insert_init(iquery)
+
+   local c_val
+   local pad_val
+
+   cursize = 1
+
+
+   for i = 1, sysbench.opt.batch_insert_count do
+
+       if (sysbench.opt.batch_insert_count <=2000) then
+           c_val = get_c_value()
+           pad_val = get_pad_value()
+       else
+           c_val = get_c_value_smaller()
+           pad_val = get_pad_value_smaller()
+       end
+
+       if (sysbench.opt.auto_inc) then
+          query = string.format("(%d, '%s', '%s')",
+                                sysbench.rand.default(1, sysbench.opt.table_size),
+                                c_val, pad_val)
+       else
+          query = string.format("(%d, %d, '%s', '%s')",
+                                sysbench.opt.table_size+pk_number,
+                                sysbench.rand.default(1, sysbench.opt.table_size),
+                                c_val, pad_val)
+          pk_number=pk_number+1
+       end
+
+       if (cursize % sysbench.opt.batch_insert_count ~= 0) then
```

Comment:
Why is this check needed? The "for" loop runs for batch_insert_count times. We can do con:bulk_insert_done() outside of for loop and con:bulk_insert_next(query) inside the for loop